File: couch/php/delete/couch_delete.php
#! /usr/bin/php
<?php
// ------------------------------------------------------------------
//	php/delete/couch_delete.php
//
//					Jun/20/2014
//
// ------------------------------------------------------------------
$path="/var/www/data_base/common/php_common";
set_include_path (get_include_path() . PATH_SEPARATOR . $path);
//
include "curl_get.php";

// ------------------------------------------------------------------
$key_in = $argv[1];

print	"*** 開始 ***\n";

print	"key_in = " . $key_in . "\n";

$url_in = 'http://localhost:5984/nagano';
$url_target = $url_in . "/" . $key_in;
print	"url_in = " .$url_in . "\n";
//
$json_string = curl_get_proc ($url_target);
print $json_string . "\n";
$unit_aa = json_decode ($json_string,true);

if (isset ($unit_aa['_rev']))
	{
	print $unit_aa['name'] . "\n";
	print $unit_aa['_rev'] . "\n";

	$url_del = $url_target . "?rev=" . $unit_aa['_rev'];

	curl_delete_proc ($url_del);
	}

print	"*** 終了 ***\n";

// ------------------------------------------------------------------
?>

File: couch/perl/delete/couch_delete.pl
#! /usr/bin/perl
#
#	couch_delete.pl
#
#					Jun/23/2014
# -------------------------------------------------------
use	strict;
use	warnings;
use	Encode;
use	JSON;
#
use lib '/var/www/data_base/common/perl_common';
use file_io;
use text_manipulate;
use uri_get;
#
# -------------------------------------------------------
print "*** 開始 ***\n";
#
my $key_in = $ARGV[0];
print ("$key_in\n");
#
my $url = 'http://localhost:5984/nagano';
#
my $url_target = $url . "/" . $key_in;
#
my $str_json = uri_get::uri_get_proc ($url_target);
#
my $data = decode_json ($str_json);
#
#
my %datax = %{$data};
#
if (exists $datax{"_rev"})
	{
	print "*** exists ***\n";
my $name = $data->{name};
my $rev = $data->{_rev};
print $key_in . "\t";
print encode ('utf-8',$name) . "\t";
print $rev . "\n";
	my $url_del = $url_target . '?rev=' . $rev;
	uri_get::uri_delete_proc ($url_del);
	}
else
	{
	print "*** not exist ***\n";
	}
#if (exists $data{$key_in})
#	{
#	uri_get::uri_put_proc ($uri,$json_str_new);
#	}
#
print "*** 終了 ***\n";
#
# -------------------------------------------------------
